packages/keyring-utils/src/btc/address.ts

```diff
@@ -1,4 +1,17 @@
-import { validate, Network } from 'bitcoin-address-validation';
+import type { AddressType } from 'bitcoin-address-validation';
+import { validate, Network, getAddressInfo } from 'bitcoin-address-validation';
+
+/**
+ * Returns whether an address is on the Bitcoin mainnet.
+ *
+ * @param address - The address to check.
+ * @param type - The address type to check.
+ * @returns `true` if the address is valid, `false` otherwise.
+ */
+export function isBtcAddress(address: string, type: AddressType): boolean {
```

Comment:
Since we're "leaking" the `AddressType` from `bitcoin-address-validation`, I wonder if we should have this function in `packages/keyring-api/src/btc/types.ts` directly, and this one would just become a `validate(address)` (without knowledge of the address's type)?

an other alternative would be to use `type: 'p2sh' | 'p2wkph' | ...` and cast it to `AddressType` (this could be slightly unsafe though..)

Reply:
true, I'll move it to the `types.ts` as a private function. That's what I initially did and then saw that similar functions were exported from keyring-utils.